samza-core/src/main/java/org/apache/samza/operators/spec/SinkOperatorSpec.java

```diff
@@ -44,45 +50,36 @@
   private final int opId;
 
   /**
-   * The user-defined sink function
+   * The sink function to send messages
    */
   private final SinkFunction<M> sinkFn;
 
-  /**
-   * Potential output stream defined by the {@link SinkFunction}
-   */
-  private final OutputStream<M> outStream;
 
   /**
-   * Default constructor for a {@link SinkOperatorSpec} w/o an output stream. (e.g. output is sent to remote database)
+   * Constructs a {@link SinkOperatorSpec} with a user defined {@link SinkFunction}.
    *
    * @param sinkFn  a user defined {@link SinkFunction} that will be called with the output message,
    *                the output {@link org.apache.samza.task.MessageCollector} and the
    *                {@link org.apache.samza.task.TaskCoordinator}.
-   * @param opCode  the specific {@link OpCode} for this {@link SinkOperatorSpec}. It could be {@link OpCode#SINK}, {@link OpCode#SEND_TO},
-   *                or {@link OpCode#PARTITION_BY}
-   * @param opId  the unique id of this {@link SinkOperatorSpec} in the {@link org.apache.samza.operators.StreamGraph}
+   * @param opCode  the specific {@link OpCode} for this {@link SinkOperatorSpec}.
+   *                It could be {@link OpCode#SINK}, {@link OpCode#SEND_TO}, or {@link OpCode#PARTITION_BY}
+   * @param opId  the unique ID of this {@link SinkOperatorSpec} in the {@link org.apache.samza.operators.StreamGraph}
```

Comment:
Question: is SAMZA-1101 included in this patch or later? If included, the sink function to actually send to an output stream should not be directly coded and instantiated in SinkOperatorSpec(). The OutputOperatorSpec should not directly include SinkFunction() { collector.send(...); }, if we want to allow dynamic instantiation of different send functions to different Stream implementation in various different runtime environment. Is it something that you plan to do by creating different collector instance from the environment?